tests/e2e/policy_driven_vol_allocation.go

```diff
@@ -1374,6 +1377,635 @@ var _ = ginkgo.Describe("[vol-allocation] Policy driven volume space allocation
+
+	/*
+		Verify expansion during Thin -> EZT, LZT -> EZT conversion (should take >vpxd task timeout)
+		Steps for offline volumes:
+		1	Create 2 SPBM policies with thin, and LZT volume allocation respectively, say pol1, pol2
+		2	Create 2 SCs each with a SPBM policy created from step 1
+		3	Create a PVC of 10g using each of the SCs created from step 2
+		4	Wait for PVCs created in step 3 to be bound
+		5	Change volume allocation of pol1 and pol2 to EZT and opt for immediate updation
+		6	While updation in step 5 is still running, expand all PVCs created in step 3 such that resize should take
+		    more than vpxd task timeout for PVCs updated to EZT allocation
+		7	Wait for all PVCs created in step 3 to reach FileSystemResizePending state
+		8	Delete the PVCs created in step 3
+		9	Delete the SCs created in step 2
+		10	Deleted the SPBM policies created in step 1
+
+		Steps for online volumes:
+		1	Create 2 SPBM policies with thin, and LZT volume allocation respectively, say pol1, pol2
+		2	Create 2 SCs each with a SPBM policy created from step 1
+		3	Create a PVC of 10g using each of the SCs created from step 2
+		4	Wait for PVCs created in step 3 to be bound
+		5	Create pods using PVCs created in step 4
+		6	Change volume allocation of pol1 and pol2 to EZT and opt for immediate updation
+		7	While updation in step 5 is still running, expand all PVCs created in step 3 such that resize should take
+		    more than vpxd task timeout for PVCs updated to EZT allocation
+		8	Wait for file system resize to complete on all PVCs created in step 3
+		9	Delete pods created in step 4
+		10	Delete the PVCs created in step 3
+		11	Delete the SCs created in step 2
+		12	Deleted the SPBM policies created in step 1
+	*/
+	ginkgo.It("[csi-block-vanilla] Verify expansion during Thin -> EZT, LZT -> EZT conversion"+
+		" (should take >vpxd task timeout)", func() {
+
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sharedvmfsURL := os.Getenv(envSharedVMFSDatastoreURL)
+		if sharedvmfsURL == "" {
+			ginkgo.Skip(fmt.Sprintf("Env %v is missing", envSharedVMFSDatastoreURL))
+		}
+
+		scParameters := make(map[string]string)
+		policyNames := []string{}
+		policyIds := []*pbmtypes.PbmProfileId{}
+		pvcs := []*v1.PersistentVolumeClaim{}
+		scs := []*storagev1.StorageClass{}
+		pvcs2d := [][]*v1.PersistentVolumeClaim{}
+		largeSize := os.Getenv(envDiskSizeLarge)
+		if largeSize == "" {
+			largeSize = diskSizeLarge
+		}
+
+		rand.New(rand.NewSource(time.Now().UnixNano()))
+		suffix := fmt.Sprintf("-%v-%v", time.Now().UnixNano(), rand.Intn(10000))
+		categoryName := "category" + suffix
+		tagName := "tag" + suffix
+
+		catID, tagID := createCategoryNTag(ctx, categoryName, tagName)
+		defer func() {
+			deleteCategoryNTag(ctx, catID, tagID)
+		}()
+
+		attachTagToDS(ctx, tagID, sharedvmfsURL)
+		defer func() {
+			detachTagFromDS(ctx, tagID, sharedvmfsURL)
+		}()
+
+		allocationTypes := []string{
+			thinAllocType,
+			lztAllocType,
+		}
+
```

Comment:
Please add relevant code for WCP and GC for it to work.

Reply:
I don't have a GC testbed handy, will add that it a subsequent PR.